### PR TITLE
RD-5607 Add an option to customize the first screen messaging

### PR DESCRIPTION
**Pull request by leon-good-life**
## Description
Jira ticket: RD-5607

I implemented according to the following comment:
https://cloudifysource.atlassian.net/browse/RD-5607?focusedCommentId=77330  Here is the copy of the comment:
> Jakub Niezgoda 
> August 22, 2022, 4:27 AM
> @ Leon Gilyadov You will probably need to change the format of schema files (cloudSetup.schema.json and  gettingStarted.schema.json). Currently they are list of objects and if we add custom message for each file, then more suitable would be to have something like:
> ```
> {
>     welcomeText: "...",
>     content: <current file content>
> }
>```
> The current text is in en.json file at the following path: /gettingStartedModal/welcomeText

## Screenshots / Videos
Changing `welcomeText` to `"aaa"`:
```
{
    "welcomeText": "aaa",
    "content": [
        {
            "name": "aws",
...
```
resulting in:
Screen Shot 2022-10-16 at 11 33 48

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
16-Oct-2022 `getting_started_spec` ✅  passed:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3698/pipeline

 

## Documentation
There is no any documentation about Getting Started Modal.

### CODE REVIEW

File: app/components/GettingStartedModal/schema/gettingStarted.schema.json [hunk 1]
```diff
@@ -1,116 +1,119 @@
...
-      },
-      {
-        "id": "terraform_aws_vm",
-        "name": "Terraform-AWS-VM",
-        "zipUrl": "https://repository.cloudifysource.org/cloudify/blueprints/6.4/terraform/terraform_aws_vm.zip",
-        "yamlFile": "blueprint.yaml"
-      }
+{
+    "welcomeText": "Welcome to Cloudify: your open source,\nmulti-cloud orchestration platform.\n\nWe’ll start by setting up your cloud environment\nin the following easy steps. Ready? Let’s go.\n\nNote: this process can take up to 5 minutes,\nand you will need your API access keys.",
+    "content": [
```
Comment (leon-good-life) [inline app/components/GettingStartedModal/schema/gettingStarted.schema.json:3]: This is the only part that changed in this file apart of code formatting. 
There are also automatic indents that was added. If thats wrong I can change the configuration in the IDE. I guess we can use Prettier library, so every UI team member will have the same configuration.

File: app/components/GettingStartedModal/schema/cloudSetup.schema.json [hunk 2]
```diff
@@ -1,375 +1,378 @@
...
-             "name": "VM-vSphere-CFY",
-             "zipUrl": "https://repository.cloudifysource.org/cloudify/blueprints/6.4/vsphere/vsphere_vm.zip",
-             "yamlFile": "blueprint.yaml"
-          }
-       ]
-    }
- ]
+{
+    "welcomeText": "Welcome to Cloudify: your open source,\nmulti-cloud orchestration platform.\n\nWe’ll start by setting up your cloud environment\nin the following easy steps. Ready? Let’s go.\n\nNote: this process can take up to 5 minutes,\nand you will need your API access keys.",
+    "content": [
```
Comment (leon-good-life) [inline app/components/GettingStartedModal/schema/cloudSetup.schema.json:3]: This is the only part that changed in this file apart of code formatting. 
There are also automatic indents that was added. If thats wrong I can change the configuration in the IDE. I guess we can use Prettier library, so every UI team member will have the same configuration.

File: app/components/GettingStartedModal/steps/WelcomeStep.tsx [hunk 3]
```diff
@@ -4,10 +4,15 @@ import StageUtils from '../../../utils/stageUtils';
 import LogoLabel from '../../banner/LogoLabel';
 import LargeLogo from '../../banner/LargeLogo';
 import renderMultilineText from '../../../utils/shared/renderMultilineText';
+import type { GettingStartedSchema } from '../model';
 
 const t = StageUtils.getT('gettingStartedModal');
 
-const WelcomeStep = () => {
+type Props = {
```
Comment (kubama) [inline app/components/GettingStartedModal/steps/WelcomeStep.tsx:11]:
```suggestion
type WelcomeStepProps = {
```

Comment (leon-good-life) [reply app/components/GettingStartedModal/steps/WelcomeStep.tsx]: I changed to `WelcomeStepProps`: [8bc03d6](https://github.com/cloudify-cosmo/cloudify-stage/pull/2218/commits/8bc03d64467b69060fb7ad3a4dcda822fd74814b)

File: app/components/GettingStartedModal/ModalContent.tsx [hunk 4]
```diff
@@ -54,7 +54,7 @@ const ModalContent = ({
 
     return (
         <Modal.Content style={{ minHeight: 220, flexDirection: 'column' }}>
-            {stepName === StepName.Welcome && <WelcomeStep />}
+            {stepName === StepName.Welcome && <WelcomeStep schema={schema} />}
```
[the remark this answers is not in the document]
Comment (leon-good-life) [reply app/components/GettingStartedModal/ModalContent.tsx:57]: I created a follow-up PR: #2220
I moved the ticket back to in-progress in Jira.

[the remark this answers is not in the document]
Comment (qooban) [reply app/components/GettingStartedModal/ModalContent.tsx:57]: Thanks.

**Pull request merged**